chains/manager.go

```diff
@@ -1569,3 +1592,238 @@ func (m *manager) getOrMakeVMGatherer(vmID ids.ID) (metrics.MultiGatherer, error
+		return nil, err
+	}
+
+	// Asynchronously passes messages from the network to the consensus engine
+	return handler.New(
+		ctx,
+		vm,
+		vm.WaitForEvent,
+		m.Validators,
+		m.FrontierPollFrequency,
+		m.ConsensusAppConcurrency,
+		m.ResourceTracker,
+		sb,
+		connectedValidators,
+		peerTracker,
+		handlerReg,
+		halter.Halt,
+	)
+}
+
+// createMessageSender creates a sender that passes messages from the consensus engine to the network
+func (m *manager) createMessageSender(ctx *snow.ConsensusContext, sb subnets.Subnet) (common.Sender, error) {
+	msgSender, err := sender.New(
+		ctx,
+		m.MsgCreator,
+		m.Net,
+		m.ManagerConfig.Router,
+		m.TimeoutManager,
+		p2ppb.EngineType_ENGINE_TYPE_CHAIN,
+		sb,
+		ctx.Registerer,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't initialize sender: %w", err)
+	}
+
+	if m.TracingEnabled {
+		msgSender = sender.Trace(msgSender, m.Tracer)
+	}
+
+	return msgSender, nil
+}
+
+func (m *manager) createTrackedPeers(primaryAlias string) (tracker.Peers, error) {
+	stakeReg, err := metrics.MakeAndRegister(
+		m.stakeGatherer,
+		primaryAlias,
+	)
+	if err != nil {
+		return nil, err
+	}
+	connectedValidators, err := tracker.NewMeteredPeers(stakeReg)
+	if err != nil {
+		return nil, fmt.Errorf("error creating peer tracker: %w", err)
+	}
+	return connectedValidators, nil
+}
+
+// createPeerTracker creates a peer tracker for the Snowman consensus engine
+func (m *manager) createPeerTracker(ctx *snow.ConsensusContext, primaryAlias string) (*p2p.PeerTracker, error) {
+	p2pReg, err := metrics.MakeAndRegister(
+		m.p2pGatherer,
+		primaryAlias,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	peerTracker, err := p2p.NewPeerTracker(
+		ctx.Log,
+		"peer_tracker",
+		p2pReg,
+		set.Of(ctx.NodeID),
+		nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("error creating peer tracker: %w", err)
+	}
+	return peerTracker, nil
+}
+
+func simplexCtxConfig(ctx *snow.ConsensusContext) simplex.SimplexChainContext {
+	return simplex.SimplexChainContext{
+		NodeID:    ctx.NodeID,
+		ChainID:   ctx.ChainID,
+		SubnetID:  ctx.SubnetID,
+		NetworkID: ctx.NetworkID,
+	}
+}
+
+func (m *manager) createSimplexChain(ctx *snow.ConsensusContext, vm block.ChainVM, sb subnets.Subnet, genesisData []byte, fxs []*common.Fx) (*chain, error) {
+	ctx.Lock.Lock()
+	defer ctx.Lock.Unlock()
+
+	ctx.State.Set(snow.EngineState{
+		Type:  p2ppb.EngineType_ENGINE_TYPE_CHAIN,
+		State: snow.Initializing,
+	})
+
+	primaryAlias := m.PrimaryAliasOrDefault(ctx.ChainID)
+	m.Log.Info("creating simplex chain", zap.String("chain", primaryAlias))
+
+	messageSender, err := m.createMessageSender(ctx, sb)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't create snowman message sender: %w", err)
+	}
+
+	// initialize the VM
+	vmDB, simplexDB, err := m.createSimplexDBs(primaryAlias, ctx.ChainID)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't create simplex DBs: %w", err)
+	}
+
+	// TODO: wrap the vm in a LOCKED VM that locks the VM interface methods
```

Comment:
I don't think we need this change notifier. I made it so that the snowman VM will re-subscribe to the events when the preference changes.

The reason it's needed for snowman is because of the proposerVM - the proposerVM's subscription to the VM is dumb so it may indefinitely wait for an event from the VM and never know it needs to re-subscribe again. The reason is that when you wait for an event in the proposerVM, it computes the time needed to sleep and then waits until that time.
However, as blocks change, that time may change dynamically, so this change notifier detects this and asks the wait for event to re-compute the wait time (yeah I know, it's an incredibly awkward mechanism, kind of a rube Goldberg machine).

In contrast, Simplex doesn't need this.